tools/atip/atip/web/web.py

```diff
@@ -477,6 +481,83 @@ def click_element_coords(self, x, y, key, display=True):
             return True
         return False
 
+    def execute_js_code(self, js_code):
+        try:
+           return self.driver.execute_script(js_code)
+        except Exception as e:
+           print "Execute js code failed: %s" % e
+           return 0   
+ 
+    #Calculate the location params of element
+    def calculate_element_location(self, key, width=0, height=0):
+        try:
+           if width:
+              width = string.atoi(width)
+           if height:
+              height = string.atoi(height)
+           js_script = 'var top=document.getElementById(\"' + key + '\").getBoundingClientRect().top;  return top'
+           top = self.execute_js_code(js_script)
+           js_script = 'var left=document.getElementById(\"' + key + '\").getBoundingClientRect().left; return left'
+           left = self.execute_js_code(js_script)
+           if not width:
+               js_script = 'var width=document.getElementById(\"' + key + '\").getBoundingClientRect().width; return width'
+               width = self.execute_js_code(js_script)
+           if not height:
+               js_script = 'var height=document.getElementById(\"' + key + '\").getBoundingClientRect().height; return height'
+               height = self.execute_js_code(js_script)
+           return (left, top, left + width, top+height)
+        except Exception as e:
+           print "Get element location failed: %s" % e
+           return 0
+
+    def calculate_resolution_ratio(self, pic_name):
+        try:
+           js_script = 'var width=window.screen.availWidth; return width'
+           body_width = self.execute_js_code(js_script)
+           js_script = 'var height=window.screen.availHeight; return height'
+           body_height = self.execute_js_code(js_script)
+           im = Image.open(pic_name)
+           w,h = im.size
+           ratio_w = w/body_width
+           ratio_h = h/body_height
+           ration = 0
+           if ratio_w > ratio_h:
+               ratio = ratio_w
+           else:
+               ratio = ratio_h
+           return w/ratio,h/ratio
+        except Exception as e:
+           print "Calculate page picture resolution failed: %s" % e
+           return 0 
+             
+    #Save the specified element as a single picture
+    def save_div_as_picture(self, key, element_pic, width=0, height=0):
+        try:
+           page_pic = "page.png"
+           self.driver.get_screenshot_as_file(page_pic)
+           self.picture_list.append(page_pic)
+           ratio = self.calculate_resolution_ratio(page_pic)
+           self.convert_pic(page_pic, ratio)
+           box = self.calculate_element_location(key, width, height)
+           self.crop_pic(page_pic, element_pic, box)
```

Comment:
Still self? You have moved "crop_pic" to common folder. Please also update others. We should have unit test for each new added function.